### PR TITLE
pacote: add missing `packumentCache` option

### PR DESCRIPTION
**Pull request by ljharb**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [ ] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [ ] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [ ] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/npm/pacote/blob/80cce46e807026e7d4ad750620c00e62539b0e46/lib/registry.js#L24-L30

### CODE REVIEW

**Comment by ljharb** [comment]
cc @peterblazejewicz got another one for you when you have a sec <3

File: types/pacote/index.d.ts [hunk 1]
```diff
@@ -249,6 +249,16 @@ export interface PacoteOptions {
...
+
+    /**
+     * you usually don't want to fetch the same packument multiple times in
+     * the span of a given script or command, no matter how many pacote calls
+     * are made, so this lets us avoid doing that.  It's only relevant for
+     * registry fetchers, because other types simulate their packument from
+     * the manifest, which they memoize on this.package, so it's very cheap
+     * already.
+     */
+    packumentCache?: Map<string, Packument>;
```
Comment (peterblazejewicz) [inline types/pacote/index.d.ts:261]:
```suggestion
    packumentCache?: Map<string, Packument> | undefined;
```
IMO it should work with exact optional type flag, as it uses expression evaluation in the source (`this.opts.packumentCache || null`)

**Comment by DangerBotOSS** [comment]
Inspecting the JavaScript source for this package found some properties that are not in the .d.ts files.
The check for missing properties isn't always right, so take this list as advice, not a requirement.
## braintree-web ([<kbd>unpkg</kbd>](https://unpkg.com/browse/braintree-web@latest/))
was missing the following properties:
1. masterpass
1. paymentRequest
1. visaCheckout
1. preferredPaymentMethods
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against facc2b1d1814d57a33a9ff31471ed5339fdada9e
</p>

**Comment by ljharb** [comment]
um, what is braintree-web

**Review by peterblazejewicz — APPROVED** [review]
LGTM!
@ljharb thx!

**Comment by ljharb** [comment]
Ready to merge

**Pull request merged**